src/FieldWarning/Assets/Terrain/Scripts/TerrainMap.cs

```diff
@@ -129,94 +138,109 @@ public TerrainMap(Terrain[] terrains1D)
             }
         }
 
-        // get our bridge positions so we dont have to be in the main thread to access bridge into
 
+        // get our bridge positions so we dont have to be in the main thread to access bridge into
         for (int i = 0; i < _bridges.Length; i++)
         {
             GameObject bridge = _bridges[i];
             _bridgePositions.Add(bridge.transform.position);
         }
 
-        _loader = new Loading("Terrain");
+        //_loader = new Loading("Terrain");
 
-        //TODO create some debug UI to dump the map when needed
         // TODO need to also somehow verify the height map is valid?? 
         // not sure how to do this each time without reading the original data.
         if (!File.Exists(_HEIGHT_MAP_PATH))
         {
-            _map = CreateHeightMapFromtTerrainData();
-            WriteWaterMap(_map, _HEIGHT_MAP_PATH);
+
+
+            // this cannot be in another thread because it uses the terrain and cannot cache the terrain because
+            // it is just as slow, so no point in putting it in a worker. Nothing we can really do about this 
+            // unless one day we release every map with it's compressed data.
+            //_loader.AddWorker(LoadWater, null, true, "Loading water");
+
+            //LoadingScreen.SWorkers.Enqueue(new CoroutineWorker(LoadWaterRunner, "Loading water."));
+
+            AddCouroutine(LoadWaterRunner, "Load water");
+
+            // Loading bridges from a separate thread throws an exception.
+            // This is why we first cache the bridge positions outside the thread
+            // before doing the below. same goes for roads and trees.
+            AddMultithreadedRoutine(LoadTreesRunner, "Loading trees.");
+            AddMultithreadedRoutine(LoadRoadsRunner, "Loading roads.");
+            AddMultithreadedRoutine(LoadBridgesRunner, "Loading bridges.");
+            AddMultithreadedRoutine(ExportFinishedMapRunner, "Creating Compressed Map File");
+
+
         } else
         {
-            _loader.AddWorker(LoadWaterMap, "Placing water");
+            //_loader.AddWorker(null, LoadCompressedMapRunner, false, "Reading Compressed Map Data");
+            AddMultithreadedRoutine(LoadCompressedMapRunner, "Reading Compressed Map Data");
         }
 
 
-        // Loading bridges from a separate thread throws an exception.
-        // This is why we first cache the bridge positions outside the thread
-        // before doing the below. same goes for roads and trees.
-        _loader.AddWorker(LoadTrees, "Setting tree positions");
-        _loader.AddWorker(LoadRoads, "Connecting roads");
-        _loader.AddWorker(LoadBridges, "Loading bridges");
-       
         // leave this commented out until we make a change and need to retest the map
         //_loader.AddWorker(MapTester);
 
     }
 
-    private void MapTester()
+    private string getMapCachePath()
```

Comment:
GetMapCachePath

Reply:
this is not really a path for height maps.. just general terrain maps.. Currently we dont generate a height map, only a terrain map

Reply:
GetTerrainMapCachePath then :)